### PR TITLE
[Yaml] fix parsing some block sequences

### LINKED ISSUE #40438: Yaml parser incorrectly parses some yaml block sequences
**Symfony version(s) affected**: 5.x-dev

**Description**
Symfony Yaml parser incorrectly parses some block sequences.
In particular, if using [this test data](https://github.com/yaml/yaml-test-suite/blob/115c4ac2b9b2e7929632da1444fd069b48f7a31f/test/3ALJ.tml), the output produced is

```json
[
  "- s1_i1 - s1_i2",
  "s2"
]
```

which is not the expected output as per the test.

**How to reproduce**
```php
<?php
require_once '../vendor/autoload.php';

$yaml = <<<Y
- - s1_i1
  - s1_i2
- s2
Y;


echo(json_encode(\Symfony\Component\Yaml\Yaml::parse($yaml)));
```

**Additional context**
Related issues:
* https://github.com/craftcms/cms/issues/5777
* https://github.com/prettier/prettier/issues/10506

### PR DESCRIPTION
**Pull request by a1812**
| Q             | A
| ------------- | ---
| Branch?       | 4.4 
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| Tickets       | Fix #40438
| License       | MIT
| Doc PR        |

### CODE REVIEW

File: src/Symfony/Component/Yaml/Parser.php [hunk 1]
```diff
@@ -556,18 +559,19 @@ private function getCurrentLineIndentation(): int
...
+     * @param int|null $indentation        The indent level at which the block is to be read, or null for default
+     * @param bool     $inSequence         True if the enclosing data structure is a sequence
+     * @param bool     $isInlineFirstChild True if the definition has an inline first element
      *
      * @return string A YAML string
      *
      * @throws ParseException When indentation problem are detected
      */
-    private function getNextEmbedBlock(int $indentation = null, bool $inSequence = false): string
+    private function getNextEmbedBlock(int $indentation = null, bool $inSequence = false, $isInlineFirstChild = false): string
```
Comment (stloyd) [inline src/Symfony/Component/Yaml/Parser.php]:
```suggestion
    private function getNextEmbedBlock(int $indentation = null, bool $inSequence = false, bool $isInlineFirstChild = false): string
```

**Review by stloyd — CHANGES_REQUESTED** [review]

File: src/Symfony/Component/Yaml/Parser.php (hunk 1 above)
Comment (a1812) [reply src/Symfony/Component/Yaml/Parser.php]: ok, fixed

File: src/Symfony/Component/Yaml/Tests/Fixtures/YtsBasicTests.yml [hunk 2]
```diff
@@ -200,3 +200,41 @@ php: |
...
+  [
+    [
+      "s1_i1",
+      [
+        "s1_i1_1",
+        "s1_i1_2",
+      ],
+      "s1_i2"
+    ],
+      "s2"
```
Comment (OskarStark) [inline src/Symfony/Component/Yaml/Tests/Fixtures/YtsBasicTests.yml]: The indentation looks wrong

I am currently on a phone 📱🧐

Comment (a1812) [reply src/Symfony/Component/Yaml/Tests/Fixtures/YtsBasicTests.yml]: ok, fixed

**Comment by a1812** [comment]
Hi @xabbuh please check my super code :)

**Comment by xabbuh** [comment]
Wow, this looks great @a1812! 👍  I have some suggestions how to simplify the patch a bit. Please take a look at a1812/symfony#1 if you agree with them.

**Review by xabbuh — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
Thank you @a1812.

**Pull request merged**